src/mavoscript.js

```diff
@@ -29,11 +29,19 @@ var _ = Mavo.Script = {
 			if (Array.isArray(a)) {
 				result = [];
 				var max = Math.max(a.length, b.length);
+				var leftIdentity = o.leftIdentity === undefined ? o.identity : o.leftIdentity;
+				var rightIdentity = o.rightIdentity === undefined ? o.identity : o.rightIdentity;
+
 				for (let i = 0; i < max; i++) {
-					result[i] = o.scalar(
-						a[i] === undefined ? o.identity : a[i],
-						b[i] === undefined ? o.identity : b[i]
-					);
+					if (a[i] !== undefined && b[i] !== undefined) {
```

Comment:
Nitpick: You are checking if `a[i]` is defined twice. You could do something like:

```js
if (a[i] === undefined) {
// ...
}
else if (b[i] === undefined) {
// ...
}
else {
// Both are defined if we got here
}
```